### PR TITLE
Support naming customization on helm chart resources

### PR DESCRIPTION
**Pull request by fgalind1**
Some resources use `include "airflow.fullname" .` helper and some just use `{{ .Release.Name }}`. fullname helper is a common good practice to name all the different helm resources.

This allows to customize the resources and leverage the existing options `fullnameOverride` or `nameOverride` and have consistency in all resources to always use `include "airflow.fullname" .`

This makes it much easier when deploying airflow as a dependency on another chart or in the same namespace, as we can clearly see airflow's chart name "airflow" in the resources and/or allow to customize the prefix for all airflow resources

The proposed helm chart was deployed in my local environment and all names match properly between deployments, secrets, configmaps, etc.

### CODE REVIEW

**Review by hussein-awala — DISMISSED** [review]
We must pay attention that this can cause multiple resources to be recreated with different names, which may affect monitoring/logging of these resources or break something that was created outside of the chart.

**Review by potiuk — DISMISSED** [review]
I think we should have significant note about it at least. I am a little concerned about the case when the user had `fullnameOverride` set before and performs upgrade with this new chart. What would happen then? I guess the old resources would remin leaving the k8s deployment with many duplicate resources duplicated - differing by name.

I think (or maybe I am wrong) that the only way to upgrade in case someone already had fullnameOverride is to delete the old application using old chart and recreate it using the new one. Or delete the whole namespace before reinstalling.

Or am I wrong?

If I am not wrong - then If we are to implement this change we need to have a warning in the docs at least, possibly also some warning (if possible) when you override the old chart with the new and some viable path for our existing users to migrate.

**Comment by fgalind1** [comment]
thanks for your review @potiuk

>  I am a little concerned about the case when the user had `fullnameOverride` set before and performs upgrade with this new chart. 

Actually `fullnameOverride` is not really used in all resources and only in some. My PR is precisely adding consistency so that all resources honor the same logic

>  What would happen then? I guess the old resources would remin leaving the k8s deployment with many duplicate resources duplicated - differing by name.
> 
> I think (or maybe I am wrong) that the only way to upgrade in case someone already had fullnameOverride is to delete the old application using old chart and recreate it using the new one. Or delete the whole namespace before reinstalling.
> 
> Or am I wrong?

Technically speaking old resources will be deleted by helm if these are renamed, so they will just be replaces with the new resources. The only thing we need to be careful is with the PVCs created by statefulset/deployments, as that means data needs to be moved from the old PVC to the new PVC. To make this change less disruptive, I've added this change as opt-in via an option `useStandardNaming` in values.yaml which is default to False. We may consider setting that to true in the future? At least for new installations users could follow the standard naming conventions

This PR bring consistency where by default it won't honor the standard naming convention but optionally and at user's decision they can opt in for standard naming conventions

**Comment by potiuk** [comment]
> This PR bring consistency where by default it won't honor the standard naming convention (as-it has been) but optionally and at user's discretion they can opt in for standard naming conventions. This is also reflected in the unit tests, where no changes happened by default, only in the newly added tests when useStandardNaming is set to True.

I am absolutely for consistency! I love that PR in general, in case it was not clear. It's just we have to care not only about the "target" but also how to guide our users to the "target" from current state, and possibly in the way that will not overwhelm is with "I upgraded to the new Helm Chart version from the previous one as usual and all hell broke loose" kind of issues :).  Changing somethign is easy, but making it in the way that existing users wil not complain and have an easy path to follow is what I am worried about, and this is actually the GIST of that change and part of the change should be to consider the scenario, and have a ready answer for the users on what should they do.

> To make this change non-disruptive, I've added on the next commit https://github.com/apache/airflow/commit/084b66cace565d758e2dfcef0d6d50d54be06b28 this change as an opt-in via an option useStandardNaming in values.yaml which defaults to False. We may consider setting that to true in the future? At least for new installations users could follow the standard naming conventions if they opt-in this.

Yep. this is a good step I think, but I also think we need to  be a bit more specific for the migration steps. The current proposal:

> Use standard naming for all resources using airflow.fullname template
Consider removing this later and default it to true
to make this chart follow standard naming conventions using the fullname template.
For now this is an opt-in switch for backwards compatiblity to leverage the standard naming convention
and being able to use fully fullnameOverride and nameOverride in all resources
For new installations - it is recommended to set it to True to follow standard naming conventions in all resources
For existing installations, this will rename and redeploy your resources with the new names. Be aware that this will
recreate your deployment/statefulsets along with their persistent volume claims and data storage migration may be
needed to keep your old data

This is cool, but I wonder if we can somehow be a more precise od what needs to be done (or maybe even just say don't do it - delete and recreate your installation if you want to use standard naming? - this is perfectly fine IMHO). Also, maybe we can even figure out a way to find out that someone did not have `useStandardNaming` set and fail such upgrade? That would be cool if we could do it. I would love `useStandardNaming` to be on by default. I have a feeling that by deferring setting the value to some unknown future, we are hiding the problem and not solving it. I.e. someone in the future will have to solve exactly the same problem of backwards compatibility, so either we solve it now, or let's not assume it will be easier in the future. I think if we don't set it on by default and find a good way how to address upgrades (even even by failing them if the value is nost set to false) then we are throwing the problem over the fence to future developers - who will have to bite the bullet. I would very much prefer to find a solution now.

> Technically speaking old resources will be deleted by helm if these are renamed, so they will just be replaces with the new resources. The only thing we need to be careful is with the PVCs created by statefulset/deployments, as that means data needs to be moved from the old PVC to the new PVC.

All right. I was not sure of that, but I did experiment and indeed helm nicely removed the old resources and added new ones, so that is cool. While doing the experiment noticed a few issues though (and I am sure our users will also do that, so I wanted to be able to go ahead and see what kind of problems they might have).

1. I used `airflow-override" fullnameOverride  and installed airflow (from main). 

2. Here are the list of resources I got:
(got it by `helm install airflow ~/code/airflow/chart  --dry-run | grep -E "^  name: airflow|^kind:"`)

```
kind: Secret
  name: airflow-fernet-key
kind: Secret
  name: airflow-redis-password
kind: Secret
  name: airflow-broker-url
kind: Job
  name: airflow-create-user
kind: Job
  name: airflow-run-airflow-migrations
kind: ServiceAccount
  name: airflow-override-create-user-job
kind: ServiceAccount
  name: airflow-override-migrate-database-job
kind: ServiceAccount
  name: airflow-override-redis
kind: ServiceAccount
  name: airflow-override-scheduler
kind: ServiceAccount
  name: airflow-override-statsd
kind: ServiceAccount
  name: airflow-override-triggerer
kind: ServiceAccount
  name: airflow-override-webserver
kind: ServiceAccount
  name: airflow-override-worker
kind: Secret
  name: airflow-postgresql
kind: Secret
  name: airflow-airflow-metadata
kind: Secret
  name: airflow-webserver-secret-key
kind: ConfigMap
  name: airflow-airflow-config
kind: ConfigMap
  name: airflow-statsd
kind: Role
  name: airflow-pod-launcher-role
kind: Role
  name: airflow-pod-log-reader-role
kind: RoleBinding
  name: airflow-pod-launcher-rolebinding
  name: airflow-pod-launcher-role
kind: RoleBinding
  name: airflow-pod-log-reader-rolebinding
  name: airflow-pod-log-reader-role
kind: Service
  name: airflow-postgresql-hl
kind: Service
  name: airflow-postgresql
kind: Service
  name: airflow-redis
kind: Service
  name: airflow-statsd
kind: Service
  name: airflow-triggerer
kind: Service
  name: airflow-webserver
kind: Service
  name: airflow-worker
kind: Deployment
  name: airflow-scheduler
kind: Deployment
  name: airflow-statsd
kind: Deployment
  name: airflow-webserver
kind: StatefulSet
  name: airflow-postgresql
kind: StatefulSet
  name: airflow-redis
kind: StatefulSet
  name: airflow-triggerer
kind: StatefulSet
  name: airflow-worker
```

As you observed - lack of consistency. Not good.

2. I got your change (lates) and set "useStandardNaming": true. Nicely we get (as expected, and it's cool). With few small things - I tihnk `airflow-postgresgl*` and `airflow-airflow-config`  should also be updated to follow the naming.

```
(kind-airflow-python-3.8-v1.23.17:KubernetesExecutor)> helm install airflow ~/code/airflow/chart  --dry-run | grep -E "^  name: airflow|^kind:"
kind: Secret
  name: airflow-override-fernet-key
kind: Secret
  name: airflow-override-redis-password
kind: Secret
  name: airflow-override-broker-url
kind: Job
  name: airflow-override-create-user
kind: Job
  name: airflow-override-run-airflow-migrations
kind: ServiceAccount
  name: airflow-override-create-user-job
kind: ServiceAccount
  name: airflow-override-migrate-database-job
kind: ServiceAccount
  name: airflow-override-redis
kind: ServiceAccount
  name: airflow-override-scheduler
kind: ServiceAccount
  name: airflow-override-statsd
kind: ServiceAccount
  name: airflow-override-triggerer
kind: ServiceAccount
  name: airflow-override-webserver
kind: ServiceAccount
  name: airflow-override-worker
kind: Secret
  name: airflow-postgresql
kind: Secret
  name: airflow-override-airflow-metadata
kind: Secret
  name: airflow-override-webserver-secret-key
kind: ConfigMap
  name: airflow-airflow-config
kind: ConfigMap
  name: airflow-override-statsd
kind: Role
  name: airflow-override-pod-launcher-role
kind: Role
  name: airflow-override-pod-log-reader-role
kind: RoleBinding
  name: airflow-override-pod-launcher-rolebinding
  name: airflow-override-pod-launcher-role
kind: RoleBinding
  name: airflow-override-pod-log-reader-rolebinding
  name: airflow-override-pod-log-reader-role
kind: Service
  name: airflow-postgresql-hl
kind: Service
  name: airflow-postgresql
kind: Service
  name: airflow-override-redis
kind: Service
  name: airflow-override-statsd
kind: Service
  name: airflow-override-triggerer
kind: Service
  name: airflow-override-webserver
kind: Service
  name: airflow-override-worker
kind: Deployment
  name: airflow-override-scheduler
kind: Deployment
  name: airflow-override-statsd
kind: Deployment
  name: airflow-override-webserver
kind: StatefulSet
  name: airflow-postgresql
kind: StatefulSet
  name: airflow-override-redis
kind: StatefulSet
  name: airflow-override-triggerer
kind: StatefulSet
  name: airflow-override-worker
```

So far, so good.

Run `helm upgrade`. Airflow installs but does not start. 

There are some config errors `CreateContainerConfigErrors` . I do not have time to investigate, but I presume it could be connected with the configmap name wrong?

image

**Comment by fgalind1** [comment]
> This is cool, but I wonder if we can somehow be a more precise od what needs to be done (or maybe even just say don't do it - delete and recreate your installation if you want to use standard naming? - this is perfectly fine IMHO). Also, maybe we can even figure out a way to find out that someone did not have useStandardNaming set and fail such upgrade?

Makes sense.. added some notes in chart/INSTALL let me know if that could be a bit more helpful for people that wants to use standard naming conventions

> I got your change (lates) and set "useStandardNaming": true. Nicely we get better consistency (as expected, and it's cool). With few small things - I tihnk airflow-postgresgl* and airflow-airflow-config should also be updated to follow the naming.

**airflow-airflow-config** should be fixed now, config and results-backend, I've standardized config and results-backend with the rest of the logic, thus I adjusted tests for those as well, but for those 2 upgrade goes smooth and get replaced gracefully

**airflow-postgresgl** unfortunately we cannot change anything from postgres via airflow chart as this is a sub-chart, however we can pass "fullnameOverride to postgresql to influence the name we want. e.g if we want all resources to be prefixed with 'airflow-override' then we could do: 
```
> helm install airflow chart -n airflow --dry-run  --set useStandardNaming=true,fullnameOverride=airflow-override,postgresql.fullnameOverride=airflow-override-postgresql | grep -E "^  name: airflow|^kind: State" | tail -n 10
  name: airflow-override-statsd
  name: airflow-override-webserver
kind: StatefulSet
  name: airflow-override-postgresql
kind: StatefulSet
  name: airflow-override-redis
kind: StatefulSet
  name: airflow-override-triggerer
kind: StatefulSet
  name: airflow-override-worker

```

> There are some config errors CreateContainerConfigErrors . I do not have time to investigate, but I presume it could be connected with the configmap name wrong?

Should be working on the last commit, indeed it was mismatch on the config map name. I tried without useStandardNaming, with useStandardNaming and with/without fullnameOverride and they seem to be working

Thanks again for the feedback

**Comment by fgalind1** [comment]
@potiuk does it look good with the last version?

**Comment by fgalind1** [comment]
thanks for approving running the workflows - I've just fixed the static lint errors - is this https://github.com/apache/airflow/actions/runs/5006110170/jobs/8971309829 a sporadic failure? as I believe that is unrelated to the helm chart isn't it?

**Comment by fgalind1** [comment]
looks like this time static checks are happy 👍 but [Test docker-compose quick start](https://github.com/apache/airflow/actions/runs/5006695738/jobs/8972616304?pr=31066#logs) kept failing and also [InProgressDisabledPostgres11, Py3.7: Core Providers[-amazon,google] Other Providers[amazon] WWW API Always CLI Providers[google]](https://github.com/apache/airflow/actions/runs/5006695738/jobs/8972500651?pr=31066#logs) - are those known flaky runs?

**Comment by potiuk** [comment]
I restarted those - let's see. We have a few flakes starting to appear recently. Let's see where the restart leads us.

**Comment by fgalind1** [comment]
> I restarted those - let's see. We have a few flakes starting to appear recently. Let's see where the restart leads us.

yay! looks like that did the trick - thanks

**Comment by fgalind1** [comment]
any additional comments @potiuk or @hussein-awala? happy to address if any additional suggestions :)

**Comment by fgalind1** [comment]
is it good to go @potiuk ? :)

**Comment by fgalind1** [comment]
@hussein-awala / @potiuk any additional input? :)

**Comment by hussein-awala** [comment]
@fgalind1 I will not be able to test this PR this week, I will do that and add my review next week

**Comment by fgalind1** [comment]
@hussein-awala / @potiuk wonder if you may have a chance to review this PR? :) Happy to address any feedback that's needed

**Comment by fgalind1** [comment]
airflow folks sorry to bother again, but wonder if this PR could get any traction 😃

**Review by potiuk — COMMENTED** [review]
I think this is mostly for  @jedcunningham  (sorry Jed) - this is all bout the back-compat thing. We are about to release new chart soon I think and this one is kinda, somewhat breaking and the question is is it breaking too much.

File: chart/INSTALL
Comment (jedcunningham) [inline chart/INSTALL:1]: This should be in a newsfragment instead. I think we can keep a note about using `useStandardNaming` in this file though.

File: chart/templates/secrets/fernetkey-secret.yaml
Comment (jedcunningham) [inline chart/templates/secrets/fernetkey-secret.yaml]: This change isn't safe unfortunately. If users use the default generated fernet key, it'll be changed during every helm upgrade.

File: chart/templates/secrets/redis-secrets.yaml
Comment (jedcunningham) [inline chart/templates/secrets/redis-secrets.yaml:1]: Same story with these, pre-upgrade isn't safe to add.

File: chart/templates/webserver/webserver-ingress.yaml
Comment (jedcunningham) [inline chart/templates/webserver/webserver-ingress.yaml]: nit: should we just use this fullname variable for the ingress name as well?

File: tests/charts/airflow_aux/test_basic_helm_chart.py
Comment (jedcunningham) [inline tests/charts/airflow_aux/test_basic_helm_chart.py]:
```suggestion
        assert list_of_kind_names_tuples == expected
```
This feels like a better way to check? Makes sure we don't miss any new additions.

File: chart/templates/_helpers.yaml
Comment (jedcunningham) [inline chart/templates/_helpers.yaml]:
```suggestion
      {{- printf "%s-airflow-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
```

Did you intend to do this so the `airflow` part remains? Otherwise, I don't think we need to keep this. I think all of the old objects with "airflow" in the name can be renamed safely.

Comment (fgalind1) [reply chart/templates/_helpers.yaml]: if we add -airflow-" we endup with double airflow e.g. `release-name-airflow-airflow-metadata` as Chart.Name is already airflow - that's why I didn't kept airflow there

File: chart/INSTALL
Comment (fgalind1) [reply chart/INSTALL:1]: >  newsfragment instead

do you have any suggestion where this should be or what do you mean by that? :) happy to move/amend wherever needed 😃

File: chart/templates/secrets/fernetkey-secret.yaml
Comment (fgalind1) [reply chart/templates/secrets/fernetkey-secret.yaml]: if the name changes it will never be installed anymore :( - do you have any suggestion? maybe an option in values.yaml like reinstallFernetKey? which adds the pre-upgrade hook optional? otherwise we'll never be able to update the Secret and all the references of volume mounts will be broken. that was the reason why I added it here but I'm open for suggestions :) Or we can keep these secrets as they were with the old names?

File: tests/charts/airflow_aux/test_basic_helm_chart.py
Comment (fgalind1) [reply tests/charts/airflow_aux/test_basic_helm_chart.py]: good call - addressed in https://github.com/apache/airflow/pull/31066/commits/5fb9f8e05b9c55516ba6a161775892dc396a6335

File: chart/templates/webserver/webserver-ingress.yaml
Comment (fgalind1) [reply chart/templates/webserver/webserver-ingress.yaml]: good one - addressed in next commit https://github.com/apache/airflow/pull/31066/commits/5fb9f8e05b9c55516ba6a161775892dc396a6335

**Review by fgalind1 — COMMENTED** [review]
thanks @jedcunningham for the helpful review

File: chart/INSTALL
Comment (jedcunningham) [reply chart/INSTALL:1]: Sorry, I should have given you more details. You can read about how they work [here](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#step-4-prepare-pr), just make sure to get it in the right folder. This would be a significant change, which allows you to have a summary and body.

File: chart/templates/secrets/fernetkey-secret.yaml
Comment (jedcunningham) [reply chart/templates/secrets/fernetkey-secret.yaml]: I don't have a good suggestion unfortunately. We've considered [lookup](https://helm.sh/docs/chart_template_guide/function_list/#lookup) but never used it because of the it's behavior with template (which things like helm-diff and argocd use). Welcome to the backcompat nightmare :(

It might be worth leaving comments so when we do get around to accepting breaking changes in preparation for major version 2 of the chart, we can address it then.

File: chart/templates/_helpers.yaml
Comment (jedcunningham) [reply chart/templates/_helpers.yaml]: I guess I misunderstood the intent of this helper. I thought it existed to keep airflow in the existing object names (which, as written, we aren't, per the tests).

If we consider the non-standard naming case, we are already renaming the objects, so why don't we just start using `airflow.fullname`? Do we need to have this special helper?

File: chart/templates/secrets/fernetkey-secret.yaml
Comment (fgalind1) [reply chart/templates/secrets/fernetkey-secret.yaml]: If you're good with it.. I'll keep the secrets that use the pre-install hook for now "untouched" with the original names to avoid making this a real breaking change and will add a comment in those resources for the future to think how to address the renaming of those with the standard naming

Comment (jedcunningham) [reply chart/templates/secrets/fernetkey-secret.yaml]: Sounds good, thanks.

Comment (fgalind1) [reply chart/templates/secrets/fernetkey-secret.yaml]: kept original names for fernet-key, redis-password and broker-url. Added a note in values.yaml just as a reference

File: chart/INSTALL
Comment (fgalind1) [reply chart/INSTALL:1]: thanks - added in https://github.com/apache/airflow/commit/d659cdb8313a6487019801b573270544edf566ae

File: chart/templates/_helpers.yaml
Comment (fgalind1) [reply chart/templates/_helpers.yaml]: yeah I think we can use fullname for all instead. I've removed airflow.name and kept consistency in https://github.com/apache/airflow/commit/d659cdb8313a6487019801b573270544edf566ae 😃

**Comment by fgalind1** [comment]
This PR now is not that breaking change after last commit :1, with defaults (useStandardNaming=false), only 3 resources are renamed:
- ConfigMap {release}-airflow-config to {release}-config
- Secret {release}-airflow-metadata to {release}-metadata
- Secret {release}-airflow-result-backend to {release}-result-backend

**Comment by fgalind1** [comment]
@jedcunningham wonder if you have some further suggestions/comments? :)

**Comment by potiuk** [comment]
Not really - I think it's because you ned to rebase. 154 is pretty far behind. Lots of changes since then.

image

**Comment by fgalind1** [comment]
> Not really - I think it's because you ned to rebase. 154 commits is pretty far behind. Lots of changes since then.
> 
> image

thanks for the re-run looks like they passed but I've rebased anyway :)

**Comment by potiuk** [comment]
I am afraid there are some conflicts now that need to be solved :(

**Comment by fgalind1** [comment]
> I am afraid there are some conflicts now that need to be solved :(

@potiuk merge conflicts? looks clean on my side.. but let me rebase just in case

**Comment by fgalind1** [comment]
@jedcunningham / @potiuk folks - let me know if there is anything else to add on this one 😃

**Comment by potiuk** [comment]
Run the tests. It looks pretty cool, but I will defer to @jedcunningham and others :)

**Comment by fgalind1** [comment]
@jedcunningham any update on this one? :)

File: chart/values.yaml
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
# Note:fernet-key,redis-password and broker-url secrets don't use this logic yet,
```

nit

File: chart/values.yaml
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
```

File: chart/INSTALL
Comment (jedcunningham) [inline chart/INSTALL]: Can you move this section to the docs instead? Maybe on the homepage?

File: chart/newsfragments/31066.significant.rst
Comment (jedcunningham) [inline chart/newsfragments/31066.significant.rst]:
```suggestion
This is a new opt-in switch ``useStandardNaming``, for backwards compatibility, to leverage the standard naming convention, which allows full use of fullnameOverride and nameOverride in all resources.
```

File: chart/newsfragments/31066.significant.rst
Comment (jedcunningham) [inline chart/newsfragments/31066.significant.rst]:
```suggestion
Only the following resources will be renamed using default of ``useStandardNaming=false``:
```

File: chart/INSTALL
Comment (jedcunningham) [inline chart/INSTALL]: This part should probably also be covered in the newsfragment.

File: chart/INSTALL
Comment (fgalind1) [reply chart/INSTALL]: awesome - moved to docs/helm-chart/index.rst in the next commit

File: chart/INSTALL
Comment (fgalind1) [reply chart/INSTALL]: updated in next commit - thx

File: docs/helm-chart/index.rst
Comment (jedcunningham) [inline docs/helm-chart/index.rst]: Can you add this section on the update behavior to the newsfragment too?

File: docs/helm-chart/index.rst
Comment (jedcunningham) [inline docs/helm-chart/index.rst]:
```suggestion
It is not enabled by default as this may cause unexpected behaviours on existing installations. However you can enable it using ``useStandardNaming``:

.. code-block:: yaml

    useStandardNaming: true
```

Let's switch to yaml vs helm --set.

File: docs/helm-chart/index.rst
Comment (fgalind1) [reply docs/helm-chart/index.rst]: added in https://github.com/apache/airflow/pull/31066/commits/a1f03ecc2baf0b78fb5144735fb333e2765d0b02

File: docs/helm-chart/index.rst
Comment (fgalind1) [reply docs/helm-chart/index.rst]: updated in https://github.com/apache/airflow/pull/31066/commits/a1f03ecc2baf0b78fb5144735fb333e2765d0b02

**Comment by fgalind1** [comment]
@ jedcunningham does it look good? any additional feedback? 😃

**Comment by fgalind1** [comment]
@jedcunningham / @hussein-awala / @potiuk any feedback is appreciated :)

**Comment by potiuk** [comment]
When I think of it, I believe there is one more thing needed to get this one in. The end-to-end-tests . 

We already test our helm chart extensively - not only with "helm-tests" but also with "kubernetes_tests". If we add this change without the e2e tests, then we might break things accidentally without knowing it.

Those are run here:

https://github.com/apache/airflow/blob/main/.github/workflows/ci.yml#L1517

I think this can be done rather easily:

1) extend the GitHub Actions job matrix to add "useStandardNaming" : true/false as parameter
2) add `--use-starndard-naming` to breeze's `breeze testing k8s` relevant commands and pass the matrix parameter there

I am happy to help if you are stuck on it @fgalind1, but I think it is a necessary step to keep "useStandardNaming" on-par with the current setting and make sure it will not get broken accidentally in the future. 

It's not very complex, it just needs to dive a bit deeper in Breeze's code that runs k8s, deploy helm charts and run tests on it under the hood.

You can see the instructions on how to run the E2E tests locally. in https://github.com/apache/airflow/blob/main/TESTING.rst#typical-testing-pattern-for-kubernetes-tests  and while it might look a bit overwhelming initially, the only thing to do is to make `breeze k8s deploy-airflow` accept the new `--use-standar-naming'  flag and later make sure that the same flag used in `breeze k8s run-complete-tests` is propagated down in breeze.

Happy to help if you are stuck, but the above should be good start.

**Comment by fgalind1** [comment]
> When I think of it, I believe there is one more thing needed to get this one in. The end-to-end-tests .
> 
> We already test our helm chart extensively - not only with "helm-tests" but also with "kubernetes_tests". If we add this change without the e2e tests, then we might break things accidentally without knowing it.
> 
> Those are run here:
> 
> [`main`/.github/workflows/ci.yml#L1517](https://github.com/apache/airflow/blob/main/.github/workflows/ci.yml?rgh-link-date=2023-08-07T21%3A14%3A31Z#L1517)
> 
> I think this can be done rather easily:
> 
> 1. extend the GitHub Actions job matrix to add "useStandardNaming" : true/false as parameter
> 2. add `--use-starndard-naming` to breeze's `breeze testing k8s` relevant commands and pass the matrix parameter there
> 
> I am happy to help if you are stuck on it @fgalind1, but I think it is a necessary step to keep "useStandardNaming" on-par with the current setting and make sure it will not get broken accidentally in the future.
> 
> It's not very complex, it just needs to dive a bit deeper in Breeze's code that runs k8s, deploy helm charts and run tests on it under the hood.
> 
> You can see the instructions on how to run the E2E tests locally. in [`main`/TESTING.rst#typical-testing-pattern-for-kubernetes-tests](https://github.com/apache/airflow/blob/main/TESTING.rst?rgh-link-date=2023-08-07T21%3A14%3A31Z#typical-testing-pattern-for-kubernetes-tests) and while it might look a bit overwhelming initially, the only thing to do is to make `breeze k8s deploy-airflow` accept the new `--use-standar-naming' flag and later make sure that the same flag used in `breeze k8s run-complete-tests` is propagated down in breeze.
> 
> Happy to help if you are stuck, but the above should be good start.

thanks for your reply @potiuk. I've added that new e2e in the next commit https://github.com/apache/airflow/pull/31066/commits/757b88e384b6d40592a9b6dbe28b22053eb528a0 - let me know what you think

**Comment by potiuk** [comment]
> thanks for your reply @potiuk. I've added that new e2e in the next commit [757b88e](https://github.com/apache/airflow/commit/757b88e384b6d40592a9b6dbe28b22053eb528a0) - let me know what you think

That looks really cool. Approved workflow, let's see how it works. I think I would slightly modify it and actually add martrix of executor x naming (true/false). That  of course is more tests, but if we are to treat standardNaming as "first class citizen" it should work for all executors and those tests are not THAT long

**Comment by fgalind1** [comment]
thanks @potiuk I've updated in the last commit to use a matrix and run all combinations together with the static check failure

**Comment by potiuk** [comment]
> thanks @potiuk I've updated in the last commit to use a matrix and run all combinations together with the static check failure

Looks fabulous ! Thanks for getting the extra mile and addit it all ! It was quite a bit of "extra" stuff, but it will pay of in the long run.

**Review by potiuk — APPROVED** [review]
Looks fantastic. Approved - pending test succeeding.

**Comment by potiuk** [comment]
cc:: @jedcunningham -> I think with the e2e tests and breeze support to run "standard Naming" I thiink this one looks really, really good.

**Comment by potiuk** [comment]
BTW. You also need to add the new flag to the _configy.py dictionary @fgalind1 - see the failing Breeze unit tests. This will make `breeze k8s tests --help` output consistent with others.

**Comment by fgalind1** [comment]
> BTW. You also need to add the new flag to the _configy.py dictionary @fgalind1 - see the failing Breeze unit tests. This will make `breeze k8s tests --help` output consistent with others.

yup - looks like this https://github.com/apache/airflow/pull/31066/commits/d3f9b121dc0068b172c2fbb42cb0472cb888a431 will do it

**Review by jedcunningham — APPROVED** [review]

**Comment by fgalind1** [comment]
@jedcunningham /@potiuk can I have some help in approving the workflows?

**Comment by fgalind1** [comment]
I've just rebased as it had conflicts with a merge changed on the output-commands-hash.txt. Would you mind approving the workflows again pls?

**Comment by jedcunningham** [comment]
Thanks @fgalind1! Congrats on your first commit 🎉

**Comment by potiuk** [comment]
🎉

**Comment by fgalind1** [comment]
thanks guys for the help!

**Comment by fgalind1** [comment]
do you guys know if there are plans to release the next helm chart 1.11.0 https://github.com/apache/airflow/milestone/77? :)

**Comment by jedcunningham** [comment]
Yes, very soon now that 2.7.1 is out. I plan to start on it tomorrow or next week.

**Pull request merged**
